### PR TITLE
Glt 243 generate sample alias

### PR DESCRIPTION
**Pull request by djcooke**
OICR naming schemes added (not configured by default). When sample is saved with null alias, the alias will be generated

### CODE REVIEW

**Comment by apmasell** [comment]
:+1: That's quite the regular expression. O_o

File: pom.xml [hunk 1]
```diff
@@ -387,6 +387,13 @@
     </dependency>
 
     <dependency>
+      <groupId>org.hamcrest</groupId>
+      <artifactId>hamcrest-all</artifactId>
+      <version>1.3</version>
+      <scope>test</scope>
+    </dependency>
```
Comment (tonydebat) [inline pom.xml]: This is a duplicate inclusion. 

The existing hamcrest (line 407) can be updated to be restricted to the 'test' scope.

**Comment by tonydebat** [comment]
After merging to develop the unit test HibernateSampleDaoTest.parentNameModifiedTest fails.

**Comment by djcooke** [comment]
Fixed the duplicate dependency and the test. Test was using the sampleNamingScheme instead of default namingScheme for the name field. Default namingScheme handles the SAMXX, LIBXX... type names

**Comment by tonydebat** [comment]
:+1:

**Pull request merged**